### ISSUE TITLE
cf deployment failed with "Error: Failed to generate variable '/Bosh Lite Director/cf/blobstore_admin_users_password' from config server: Director is not configured with a config server"

### ISSUE THREAD

**Issue by hugsyhugsyhugsy** [cloudfoundry/cf-deployment#424]
I have my bosh-cli installed and bosh-lite initialized successful, then i use this command to deploy cf
```
git clone https://github.com/cloudfoundry/cf-deployment.git
cd cf-deployment
bosh-cli -n -e vbox upload-stemcell https://bosh.io/d/stemcells/bosh-warden-boshlite-ubuntu-trusty-go_agent \
bosh-cli -n -e vbox update-cloud-config iaas-support/bosh-lite/cloud-config.yml \
bosh-cli -n -e vbox -d cf deploy  cf-deployment.yml \
 -o operations/bosh-lite.yml \
 --vars-store deployment-vars.yml \
 -v system_domain=bosh-lite.com \
 -v cf_admin_password=admin \
 -v uaa_admin_client_secret=admin-secret
```
the last command failed with message
```
Task 25
06:45:43 | Preparing deployment: Preparing deployment (00:00:01)
            L Error: Failed to generate variable '/Bosh Lite Director/cf/blobstore_admin_users_password' from config server: Director is not configured with a config server

06:45:44 | Error: Failed to generate variable '/Bosh Lite Director/cf/blobstore_admin_users_password' from config server: Director is not configured with a config server

Started  Thu Mar  1 06:38:44 UTC 2018
Finished Thu Mar  1 06:45:44 UTC 2018
Duration 00:07:00

Task 25 error

Updating deployment:
  Expected task '25' to succeed but was state is 'error'

Exit code 1
```

**Comment by dsabeti** [comment]
The error tells me that the BOSH director is trying to delegate variable generation to Credhub (or some other config server), but you don't have one deployed. What's odd is that you're providing the `--vars-store` flag, which should generate those variables for you instead.

- What version of the BOSH CLI are you using? (I don't actually think this is likely to be the issue, but I'm just covering my bases.)
- Could you try deploying with the `-v cf_admin_password` and `-v uaa_admin_client_secret`? I actually would expect your deploy command to work as you intend, but I wonder if there's a bug in the BOSH CLI.

**Comment by dsabeti** [comment]
Hey @hugsyhugsyhugsy, any updates on this issue?

**Comment by hugsyhugsyhugsy** [comment]
@dsabeti thank you, after i update the BOSH CLI to the latest version, the problem solved 👍

**Comment by wildloop** [comment]
I had the same problem with the last bosh2 client `version 2.0.9-3623c61-2017-03-01T01:44:58Z` and it works with bosh3 `version 3.0.1-712bfd7-2018-03-13T23:26:43Z`.

`blobstore_admin_users_password` is the first variable in the [CF deployment manifest file](https://github.com/cloudfoundry/cf-deployment/blob/master/cf-deployment.yml#L1239). This manifest probably is no longer compatible with bosh-cli 2.0 ...

**Comment by dsabeti** [comment]
@wildloop, I've got it working with version 2.0.48 of the BOSH CLI, so I wonder if there was a bug fix sometime between 2.0.9 and 2.0.48. Could you try 2.0.48 and let me know how it goes? If it looks bad, we can try a bisect to see which version of the CLI fixes the issue and document it.

**Comment by wildloop** [comment]
@dsabeti I've got it working with version 2.0.48 of the BOSH CLI too.
It started to work properly from the version 2.0.15, with the version 2.0.14 I get the error `Failed to generate variable '/cfbd/cf/blobstore_admin_users_password' from config server: Director is not configured with a config server`

My mistake was that I found "the last" BOSH-CLI 2.0 version on this page: https://s3.amazonaws.com/bosh-cli-artifacts/ 
The versions are lexicographically ordered there, so the `bosh-cli-2.0.9-windows-amd64.exe` is after the `bosh-cli-2.0.48-darwin-amd64` ... 

https://youtu.be/TKMCd7oiHPk :)
